### PR TITLE
Dp core porting

### PR DESCRIPTION
**Pull request by michal-clapinski**
NOTE: In order for this to work, go to tools/sandesh and checkout to branch 'windows'

Ported vr_packet.h. Should work but probably further testing is required.

### CODE REVIEW

File: vRouter.vcxproj [hunk 1]
```diff
@@ -201,8 +201,8 @@
     <ClCompile>
       <WarningLevel>Level4</WarningLevel>
       <TreatWarningAsError>true</TreatWarningAsError>
-      <AdditionalIncludeDirectories>..;.;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
-      <PreProcessorDefinitions>%(PreProcessorDefinitions);NDIS_WDM=1;NDIS630=1</PreProcessorDefinitions>
+      <AdditionalIncludeDirectories>..;.;C:\dev\codilime-juniper\tools\;C:\dev\codilime-juniper\tools\sandesh\library\c;C:\dev\codilime-juniper\vrouter\windows\sandesh-temp;C:\dev\codilime-juniper\vrouter\windows;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (m-kostrzewa) [inline vRouter.vcxproj]: I think we should use relative paths here

File: include/vr_platform.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+typedef BOOLEAN bool;
+
+#define true TRUE
+#define false FALSE
+
+#define __attribute__packed__open__ __pragma( pack(push, 1) )
+#define __attribute__packed__close__ __pragma( pack(pop) )
+#define __attribute__format__open__(...) /* do nothing */
+#define __attribute__format__close__(...) /* do nothing */
```
Comment (m-kostrzewa) [inline include/vr_platform.h]: Did you verify if byte alignments are the same?

File: include/vr_defs.h [hunk 3]
```diff
@@ -7,10 +7,14 @@
 #ifndef __VR_DEFS_H__
 #define __VR_DEFS_H__
 
+#include "vr_platform.h"
+
 #define MAC_FORMAT     "%02x:%02x:%02x:%02x:%02x:%02x"
 #define MAC_VALUE(x)   (x)[0],(x)[1],(x)[2],(x)[3],(x)[4],(x)[5]
 
+#ifndef _WINDOWS
 #define ARRAYSIZE(x) (sizeof(x) / sizeof((x)[0]))
```
Comment (michal-clapinski) [inline include/vr_defs.h]: ARRAYSIZE is already defined on Windows.

**Comment by michal-clapinski** [comment]
I didn't port vr_packet.c.

File: include/vr_platform.h (hunk 2 above)
Comment (michal-clapinski) [inline include/vr_platform.h]: Yeah, and?

Comment (m-kostrzewa) [inline include/vr_platform.h]: @Attil could you clarify?

Comment (Attil) [inline include/vr_platform.h]: That attribute means this function is printf-like when it comes to syntax. It allows to check correctness of the string (like `("This is my string: %s and this is my number: %u\r\n", "abc", 5);`)

Comment (m-kostrzewa) [inline include/vr_platform.h]: ah ok you're talking about *__format__*. I meant structure packing using *__packed__*

**Pull request merged**